rust/arrow/src/buffer.rs

```diff
@@ -259,16 +260,20 @@ impl Buffer {
     /// Returns a slice of this buffer starting at a certain bit offset.
     /// If the offset is byte-aligned the returned buffer is a shallow clone,
     /// otherwise a new buffer is allocated and filled with a copy of the bits in the range.
-    pub fn bit_slice(&self, offset: usize, len: usize) -> Self {
-        if offset % 8 == 0 && len % 8 == 0 {
-            return self.slice(offset / 8);
+    pub fn bit_view(&self, offset_in_bits: usize, len_in_bits: usize) -> Self {
+        if offset_in_bits % 8 == 0 && len_in_bits % 8 == 0 {
+            self.slice(offset_in_bits / 8)
```

Comment:
I don't fully understand why this doesn't need to refer to `len_in_bits` -- how do we know that `len_in_bits` covers the entire buffer? Maybe this should be `self.slice(len_in_bits/8)`?

Reply:
That is the idea, bit view doesn't cover the whole Buffer. If you give the whole buffer's length in bits and start offset as 0 then it will cover the whole buffer. Otherwise, we can use a partial bit view on the Buffer.

Reply:
> If you give the whole buffer's length in bits and start offset as 0 then it will cover the whole buffer

Right, what I don't understand is how the test for `len_in_bits % 8 == 0` is checking for the whole buffer length. It seems like it is checking that `len_in_bits` is a multiple of 8 (aka represents whole bytes)

Maybe there is some assumption here like `self.len_in_bits < 8`?